src/states/adult.rs

```diff
@@ -389,14 +388,9 @@ impl Adult {
             .map(|node| node.connection_info().clone())
             .collect();
 
-        let message = self.to_hop_message(signed_msg.clone())?;
+        let message = self.to_hop_message(signed_msg)?;
         self.send_message_to_targets(&targets, targets.len(), message);
 
-        // we've seen this message - don't handle it again if someone else sends it to us
-        let _ = self
```

Comment:
Why is that removed?

Reply:
It's redundant because it's already done in `handle_signed_message`.

Reply:
Is that enough though? that create an imbalance with same code in Elder and I'm suspicious.
What if we accumulate a signature that trigger us to send a message as an adult, will we not forget to filter it?

Reply:
We don't send any routing messages as adult right now. But even if we were, if we are the one sending, should it really go through *incoming* filter?

Reply:
We do through:

Reply:
We do through
```rust
    fn handle_message_signature(
        &mut self,
        msg: SignedRoutingMessage,
        pub_id: PublicId,
    ) -> Result<(), RoutingError> {
        if !self.chain.is_peer_elder(&pub_id) {
            debug!(
                "{} - Received message signature from not known elder (still use it) {}, {:?}",
                self, pub_id, msg
            );
        }

        if let Some(signed_msg) = self.sig_accumulator.add_proof(msg) {
            self.handle_signed_message(signed_msg)?;
        }
        Ok(())
    }
```

But then we already done the filtering so not a problem.

Reply:
> should it really go through incoming filter?

It is for Elders, and yes, so we don't process them if we receive the same message and resend them.

Reply:
Then I think the incoming filter check should happen inside `handle_message_signature` where it seems more appropriate to me. Also for elders. On the other hand, this change is not really that relevant for this PR, so I'm now leaning towards reverting it. What do you think?

Reply:
Yes, I think that would make more sense as in a different PR that clean things up if we want it.